tests/h/settings_test.py

```diff
@@ -2,99 +2,122 @@
 
 from __future__ import unicode_literals
 
+import logging
 import pytest
-import mock
 
-from h import settings
+from h.settings import (SettingsManager,
+                        SettingError,
+                        database_url)
 
 
-class FakeSetting(object):
-    def __init__(self, result):
-        self.result = result
+def asutf8(string):
+    return string.encode()
 
-    def __call__(self, environ):
-        return self.result
 
-    def __str__(self):
-        return 'fake setting'
+def test_database_url():
+    url = 'postgres://postgres:1234/database'
+    expected = 'postgresql+psycopg2://postgres:1234/database'
 
+    assert database_url(url) == expected
 
-class TestDeprecatedSetting(object):
-    def test_emits_warnings_when_child_setting_is_used(self):
-        func = settings.DeprecatedSetting(FakeSetting(result={'foo': 'bar'}),
-                                          message='what to do instead')
-        func.warn = mock.Mock(spec_set=[])
 
-        result = func({})
+class TestSettingsManager(object):
+    def test_set_does_not_warn_when_deprecated_setting_is_not_used(self, caplog):
+        with caplog.at_level(logging.WARN):
+            settings_manager = SettingsManager({}, {})
+            settings_manager.set('foo', 'FOO', deprecated_msg='what to do instead')
+        assert not caplog.records
 
-        assert result == {'foo': 'bar'}
-        func.warn.assert_called_once_with('use of fake setting is '
-                                          'deprecated: what to do instead')
+    def test_set_sets_value_when_deprecated_setting_is_used(self):
+        settings_manager = SettingsManager(environ={'FOO': 'bar'})
+        settings_manager.set('foo', 'FOO', deprecated_msg='what to do instead')
 
-    def test_emits_no_warnings_when_unused(self):
-        func = settings.DeprecatedSetting(FakeSetting(result=None),
-                                          message='what to do instead')
-        func.warn = mock.Mock(spec_set=[])
+        result = settings_manager.settings['foo']
 
-        result = func({})
+        assert result == 'bar'
 
-        assert result is None
-        assert not func.warn.called
+    def test_set_warns_when_deprecated_setting_is_used(self, caplog):
+        with caplog.at_level(logging.WARN):
+            settings_manager = SettingsManager({}, {'FOO': 'bar'})
+            settings_manager.set('foo', 'FOO', deprecated_msg='what to do instead')
+        assert 'what to do instead' in caplog.text
```

Comment:
We're missing a test that it does not warn when a non-deprecated setting is used